honeybee_schema/radiance/material.py

```diff
@@ -0,0 +1,280 @@
+    # TODO: We can't define union of schema objects that haven't been
+    # defined yet. We also can't move this down, because ModifierBase
+    # is inherited by all these objects.
+    modifier: Union[
+         Void, Plastic, Glass, BSDF, Glow, Light, Trans
+    ] = Field(
+        # TODO: This needs to be fixed. How can we add add default=Void,
+        # when Void is not serializable?
+        default='void',
+        description='Material modifier (default: Void).'
+        )
+
+    # TODO: We can't define union of schema objects that haven't been
+    # defined yet. We also can't move this down, because ModifierBase
+    # is inherited by all these objects.
+    dependencies: List[
+        Union[Void, Plastic, Glass, BSDF, Glow, Light, Trans]
+    ] = Field(
+        default=[],
+        min_items=1,
+        description='List of modifiers that this modifier depends on. '
+                    'This argument is only useful for defining advanced modifiers '
+                    'where the modifier is defined based on other modifiers '
+                    '(default: []).'
+        )
+
+
+class Plastic(ModifierBase):
+    """Radiance plastic material."""
+
+    type: constr(regex='^Plastic$') = 'Plastic'
+
+    r_reflectance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the red channel reflectance '
+                    '(default: 0).'
+    )
+
+    g_reflectance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the green channel reflectance '
+                    '(default: 0).'
+    )
+
+    b_reflectance: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the blue channel reflectance '
+                    '(default: 0).'
+    )
+
+    specularity: float = Field(
+        default=0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the fraction of specularity. '
+                    'Specularity fractions greater than 0.1 are not realistic. '
+                    '(default: 0).'
+    )
+
+    roughness: float = Field(
+        default=0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the roughness, specified as the '
+                    'rms slope of surface facets. Roughness greater than 0.2 are '
+                    'not realistic (default: 0).'
+    )
+
+
+class Trans(Plastic):
+    """Radiance Translucent material."""
+
+    type: constr(regex='^Trans$') = 'Trans'
+
+    transmitted_diff: float = Field(
+        default=0,
+        description='The fraction of transmitted light that is transmitted diffusely in '
+                    'a scattering fashion (default: 0).'
+    )
+
+    transmitted_spec: float = Field(
+        default=0,
+        description='The fraction of transmitted light that is not diffusely scattered '
+                    '(default: 0).'
+    )
```

Comment:
Is this all? If I'm not mistaken you need 7 values for a Trans material.

Reply:
Because the `Trans` material is inheriting from `Plastic`, we don't need this right now. I think it's a good use of inheritance. 

However, I think @saeranv should add a `root_validator ` to make sure that the sum of the average reflectance, `transmitted_diff` and `transmitted_spec` is less than 1. Similar to what we have here:
https://github.com/ladybug-tools/honeybee-schema/blob/master/honeybee_schema/energy/load.py#L56-L64

Reply:
@chriswmackey 

To clarify, it's `<` but not '<=` so that there's always some fraction of light being absorbed?

Reply:
Yes, I am pretty sure an absorbtance of 0 is still acceptable in Radiance world.